### PR TITLE
Switch to forked `httprouter` and enable `UseRawPath` option

### PR DESCRIPTION
**Pull request by Tener**
The [httprouter](https://github.com/julienschmidt/httprouter) package mishandles the URL-encoded `/` present in request paths. Despite requests and available PRs, this hasn't been fixed: 
- https://github.com/julienschmidt/httprouter/pull/55 
- https://github.com/julienschmidt/httprouter/pull/209

We are affected by this issue in many ways, e.g.:
- https://github.com/gravitational/teleport/pull/11011
- https://github.com/gravitational/teleport/issues/10576

This PR fixes the problem by switching to our fork of `httprouter`: https://github.com/gravitational/httprouter, which has one of the proposed patches applied. The fork's main branch is `teleport`, which is based on tag `1.3.0v` we were using.

The fork adds `UseRawPath` option to the router, which is false by default to maintain backward compatibility.

Here we enable that option for all our usages of the router. To verify the fix to our routing, a single test is added (taken from #11011).

Fixes `TEL-Q122-7`.

### CODE REVIEW

**Comment by zmb3** [comment]
I see that this will indeed fix the problem, but I can't help but ask if we should really be using httprouter at all? It's clearly no longer maintained, and each new fork we pull in creates additional burden on us.

I'm fine with merging this, but I wonder if we should also evaluate using a different router that is actively maintained.

**Comment by Tener** [comment]
> I see that this will indeed fix the problem, but I can't help but ask if we should really be using httprouter at all? It's clearly no longer maintained, and each new fork we pull in creates additional burden on us.
> 
> I'm fine with merging this, but I wonder if we should also evaluate using a different router that is actively maintained.

Indeed the `httprouter` doesn't look active right now: after 1.3.0 there was ~1 year of commits, but now it is 2 years of silence. Strangely, the PRs I linked were from 2015 and 2017 when the library was in active development. Ignoring these PRs is an argument in favor of moving away from `httprouter`.

Having said that, I'm worried about the potential costs of the switch. There may be hard-to-find compatibility issues lurking, while the benefits are seemingly few or none?

File: lib/web/apiserver_test.go [hunk 1]
```diff
@@ -2286,6 +2286,53 @@ func TestAddMFADevice(t *testing.T) {
...
+
+	totpCode, err := totp.GenerateCode(pack.otpSecret, env.clock.Now().Add(30*time.Second))
+	require.NoError(t, err)
+
+	// Obtain a privilege token.
+	endpoint := pack.clt.Endpoint("webapi", "users", "privilege", "token")
+	re, err := pack.clt.PostJSON(ctx, endpoint, &privilegeTokenRequest{
+		SecondFactorToken: totpCode,
+	})
+
```
Comment (codingllama) [inline lib/web/apiserver_test.go:2313]: nit: handle err close to declaration.

```suggestion
```

**Review by codingllama — APPROVED** [review]
Looks reasonable to me, although I didn't go all the way digging on httprouter or UseRawPath.

**Comment by codingllama** [comment]
> Having said that, I'm worried about the potential costs of the switch. There may be hard-to-find compatibility issues lurking, while the benefits are seemingly few or none?

That's true, it's a scary change to make. I think the lesson is to be very careful about the dependencies we pull in, specially far reaching ones like httprouter.

My 2c is that best way to "fix" this is to kill all REST APIs in favor of gRPC. One can dream. :)

@zmb3

**Comment by Tener** [comment]
> My 2c is that best way to "fix" this is to kill all REST APIs in favor of gRPC. One can dream. :)

There are at least two of us dreaming the same dream 😄

**Review by zmb3 — APPROVED** [review]

**Review by probakowski — APPROVED** [review]

**Pull request merged**